server/src/test/java/org/opensearch/index/store/remote/file/OnDemandBlockSnapshotIndexInputTests.java

```diff
@@ -143,6 +143,9 @@ private OnDemandBlockSnapshotIndexInput createOnDemandBlockSnapshotIndexInput(in
 
         doAnswer(invocation -> {
             BlobFetchRequest blobFetchRequest = invocation.getArgument(0);
+            if (blobFetchRequest.getLength() <= 0) {
```

Comment:
Maybe put this check when building the BlobFetchRequest? A negative length is never valid.

Reply:
Makes sense. Updated the PR.